rules/opentitan/cc.bzl

```diff
@@ -74,13 +74,22 @@ def ot_binary(ctx, **kwargs):
         compilation_contexts.append(linker_script[CcInfo].compilation_context)
 
     name = get_override(ctx, "attr.name", kwargs)
+    all_srcs = get_override(ctx, "files.srcs", kwargs)
+
+    # cc_common.compile crashes if a header file is passed to srcs, so filter
+    # those out and passed them as private headers instead
+    srcs = [s for s in all_srcs if s.extension == "c"]
+    hdrs = [s for s in all_srcs if s.extension == "h"]
```

Comment:
Its not inconceivable that we'd want to compile C++ code.  I'd structure this as:
```
hdrs = [s for s in all_srcs if file_is_a_header(s)]
srcs = [s for s in all_srcs if not file_is_a_header(s)]
```

Reply:
I changed it per your suggestion. I also overlooked that we may have linker script files in deps.